### PR TITLE
fix(replays): Fix Replay view tab links

### LINKED ISSUE #37811: Bug: Tabs (focus area & sidebar) should have `href` values set so that users can right-click and select "open in new tab"
We should drop the hash link, ie: no more `href="#console"`

Instead it should be a link to `?t_main=console` and it should maintain any other query params that might be set.

See:
- https://github.com/getsentry/sentry/blob/master/static/app/views/replays/detail/focusTabs.tsx#L26
- https://github.com/getsentry/sentry/blob/master/static/app/views/replays/detail/sideTabs.tsx#L21

### PR DESCRIPTION
**Pull request by danecando**
### Changes
- Changes tab href from hash param to full path + search params 

Closes #37811 

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: static/app/views/replays/detail/focusTabs.tsx [hunk 1]
```diff
@@ -16,23 +17,35 @@ const ReplayTabs: Record<TabKey, string> = {
 type Props = {};
 
 function FocusTabs({}: Props) {
+  const {location} = useRouteContext();
```
Comment (billyvg) [inline static/app/views/replays/detail/focusTabs.tsx:20]: Could use `useLocation()` (which wraps `useRouteContext`... not sure which is preferred, maybe grep and see what's more used).

File: static/app/views/replays/detail/focusTabs.tsx [hunk 2]
```diff
@@ -16,23 +17,35 @@ const ReplayTabs: Record<TabKey, string> = {
...
-        </li>
-      ))}
+      {Object.entries(ReplayTabs).map(([tab, label]) => {
+        const href = Object.entries({
+          ...location.query,
+          t_main: tab,
+        }).reduce(
+          (acc, [queryParam, val], i) => `${acc}${i > 0 ? '&' : ''}${queryParam}=${val}`,
+          `${location.pathname}?`
+        );
```
Comment (billyvg) [inline static/app/views/replays/detail/focusTabs.tsx:33]: This is a bit hard to grok -- I'm assuming we're trying to preserve the query string here... can we:

1) Add comments
2) Maybe use the `qs` (query-string?) library?

Comment (danecando) [reply static/app/views/replays/detail/focusTabs.tsx]: Much nicer.. I should have checked for a lib from the start

File: static/app/views/replays/detail/focusTabs.tsx [hunk 3]
```diff
@@ -16,14 +18,16 @@ const ReplayTabs: Record<TabKey, string> = {
...
   const {getActiveTab, setActiveTab} = useActiveReplayTab();
   const activeTab = getActiveTab();
+
   return (
     <NavTabs underlined>
       {Object.entries(ReplayTabs).map(([tab, label]) => (
         <li key={tab} className={activeTab === tab ? 'active' : ''}>
           <a
-            href={`#${tab}`}
+            href={`${pathname}?${queryString.stringify({...query, t_main: tab})}`}
```
Comment (billyvg) [inline static/app/views/replays/detail/focusTabs.tsx:30]: Much cleaner, thanks

**Review by billyvg — APPROVED** [review]

**Pull request merged**